test/test-platform-output.c

```diff
@@ -84,12 +84,78 @@ TEST_IMPL(platform_output) {
          (unsigned long long) rusage.ru_stime.tv_usec);
   printf("  page faults: %llu\n", (unsigned long long) rusage.ru_majflt);
   printf("  maximum resident set size: %llu\n",
-         (unsigned long long) rusage.ru_maxrss);
+         (unsigned long long)rusage.ru_maxrss);
 
   par = uv_available_parallelism();
   ASSERT_GE(par, 1);
   printf("uv_available_parallelism: %u\n", par);
 
+  #ifdef __linux__
+  FILE* file;
+  char* line = NULL;
+  size_t len = 0;
+  ssize_t read;
+  int cgroup_version = 0;
+  unsigned int cgroup_par = 0;
+
+  // Check if the system is using cgroup v2
+  file = fopen("/sys/fs/cgroup/cgroup.controllers", "r");
+  if (file != NULL) {
+    cgroup_version = 2;
+    fclose(file);
+  } else {
+    // Check if the system is using cgroup v1
+    file = fopen("/proc/self/cgroup", "r");
+    if (file != NULL) {
+      cgroup_version = 1;
+      fclose(file);
+    }
+  }
+
+  if (cgroup_version == 1) {
+    // Parse cgroup v1 to deduce parallelism constraints
+    file = fopen("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us", "r");
+    if (file) {
+      if (getline(&line, &len, file) != -1) {
+        long long quota = atoll(line);
```

Comment:
`long long` isn't exactly wrong but in libuv we generally use int64_t or uint64_t. The latter is probably more appropriate because quotas are never negative.